statsmodels/genmod/generalized_linear_model.py

```diff
@@ -208,17 +227,21 @@ def __init__(self, endog, exog, family=None, offset=None, exposure=None,
             exposure = np.log(exposure)
         if offset is not None:  # this should probably be done upstream
             offset = np.asarray(offset)
+
+        self.freq_weights = freq_weights
```

Comment:
I remember now. 

For consistency, I wanted all the shape checking to be in the `initialize` function. That is called from the `super().__init__()` call. The only argument to `initialize` is `self`, so I needed to attach early. I see your point. How do you think we should do this?